### ISSUE TITLE
1.3.1: Failed to copy file

### ISSUE THREAD

**Issue by allenfuller** [zodern/meteor-up#707]
After updating to 1.3.1 whenever I run `mup reconfig` or `mup deploy` I get the following:
```
Failed to copy file  { Error: No such file
    at SFTPStream._transform (/usr/local/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:384:27)
    at SFTPStream.Transform._read (_stream_transform.js:167:10)
    at SFTPStream._read (/usr/local/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:170:15)
    at SFTPStream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:307:12)
    at writeOrBuffer (_stream_writable.js:293:5)
    at SFTPStream.Writable.write (_stream_writable.js:220:11)
    at Channel.ondata (_stream_readable.js:556:20)
    at emitOne (events.js:96:13)
    at Channel.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
    at Channel.Readable.push (_stream_readable.js:135:10)
    at SSH2Stream.<anonymous> (/usr/local/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2/lib/Channel.js:140:15)
    at emitOne (events.js:96:13)
    at SSH2Stream.emit (events.js:188:7)
    at parsePacket (/usr/local/lib/node_modules/mup/node_modules/ssh2-streams/lib/ssh.js:3163:10) code: 2, lang: '' }
Retrying in 3 seconds
```

**Comment by zodern** [comment]
Did you run `mup setup` after updating?

**Comment by allenfuller** [comment]
That worked -- thanks! I had tried `mup reconfig` thinking that would do it. I might suggest adding that as a step in the docs.

During deployment though, I get the following error when trying to start Meteor:
```
	-----------------------------------STDERR-----------------------------------
	2-nginx-proxy
	Error response from daemon: endpoint my_app-nginx-proxy not found
	Finished removing docker containers
	--2017-08-26 20:48:33--  https://raw.githubusercontent.com/jwilder/nginx-proxy/master/nginx.tmpl
	Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
	Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
	HTTP request sent, awaiting response... 200 OK
	Length: 10529 (10K) [text/plain]
	Saving to: ‘/opt/datrminworker02/config/nginx.tmpl’
	
	     0K ..........                                            100% 11.9M=0.001s
	
	2017-08-26 20:48:33 (11.9 MB/s) - ‘/opt/my_app/config/nginx.tmpl’ saved [10529/10529]
	
	docker: Error response from daemon: driver failed programming external connectivity on endpoint my_app-nginx-proxy (6db6644936c461c8a25b7a70fed51605b51ce4f03a7d089ce2c81ca4a0c50bb4): Bind for 0.0.0.0:443 failed: port is already allocated.
```
I ended up just deleting the instance and building a new one. That allowed me to deploy successfully.

**Comment by janzheng** [comment]
edit: DigitalOcean droplet went haywire. Cheers

**Comment by Floriferous** [comment]
I'm seeing this same error on v1.3.0 and v1.3.7. Downgrading to 1.2.11 makes it work again.

The error for me happens when uploading a custom SSL certificate in the `proxy.ssl` object (here 1.3.7):
```
>mup setup

Started TaskList: Setup Docker
[139.59.211.103] - Setup Docker
[139.59.211.103] - Setup Docker: SUCCESS

Started TaskList: Setup proxy
[139.59.211.103] - Setup Environment
[139.59.211.103] - Setup Environment: SUCCESS
[139.59.211.103] - Pushing the Startup Script
[139.59.211.103] - Pushing the Startup Script: SUCCESS
[139.59.211.103] - Cleaning Up SSL Certificates
[139.59.211.103] - Cleaning Up SSL Certificates: SUCCESS
[139.59.211.103] - Copying SSL Certificate Bundle
Failed to copy file  { Error: No such file
    at SFTPStream._transform (/usr/local/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2-streams/lib/sftp.js:384:27)
    at SFTPStream.Transform._read (_stream_transform.js:167:10)
    at SFTPStream._read (/usr/local/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2-streams/lib/sftp.js:170:15)
    at SFTPStream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:334:12)
    at writeOrBuffer (_stream_writable.js:320:5)
    at SFTPStream.Writable.write (_stream_writable.js:247:11)
    at Channel.ondata (_stream_readable.js:555:20)
    at emitOne (events.js:96:13)
    at Channel.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Channel.Readable.push (_stream_readable.js:134:10)
    at SSH2Stream.<anonymous> (/usr/local/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2/lib/Channel.js:140:15)
    at emitOne (events.js:96:13)
    at SSH2Stream.emit (events.js:188:7)
    at parsePacket (/usr/local/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2-streams/lib/ssh.js:3163:10) code: 2, lang: '' }
Retrying in 3 seconds
```

And my mup config (the goal being to have 2 apps on the same VPS):
```js
module.exports = {
  servers: {
    one: {
      host: 'xxxxxx',
      username: 'root',
      pem: '~/.ssh/id_rsa',
    },
  },

  meteor: {
    name: 'my-app',
    path: 'path/to/myApp',

    servers: {
      one: {
        env: {},
      },
    },

    buildOptions: {
      serverOnly: true,
    },

    env: {
      ROOT_URL: 'https://subdomain1.my-app.com',
      MONGO_URL: 'xxx',
    },

    docker: {
      image: 'abernix/meteord:node-8-base',
    },

    enableUploadProgressBar: true,
    deployCheckWaitTime: 120,
  },

  proxy: {
    domains: 'subdomain1.my-app.com',
    ssl: {
      crt: './bundle.crt',
      key: './privatekey.key',
    },
  },
};
```

**Comment by zodern** [comment]
Starting in mup 1.4, mup will suggest running `mup setup` when this error happens.

@Floriferous the bug with setting up the proxy is fixed in mup 1.4.2.
